### ISSUE TITLE
Issue with predict() in Implementing a custom kernel in GPyTorch 

### ISSUE THREAD

**Issue by mboedigh** [cornellius-gp/gpytorch#1456]
Seems like predict() function in [Implementing a custom kernel in GPyTorch](https://github.com/cornellius-gp/gpytorch/blob/master/examples/00_Basic_Usage/Implementing_a_custom_Kernel.ipynb) should operate in eval mode.
Otherwise, using predict() on on an untrained model (e.g. after torch.load() is used to load saved parameters) will give an error

should this: 
```
def predict(model, likelihood, test_x=torch.linspace(0, 1, 51)):
    # Make predictions by feeding model through likelihood
    with torch.no_grad(), gpytorch.settings.fast_pred_var():
        # Test points are regularly spaced along [0,1]
        return likelihood(model(test_x))
```
be this (or better yet save the current eval state and return to model to the previous state
```
def predict(model, likelihood, test_x=torch.linspace(0, 1, 51)):
    #set to eval mode
    model.eval()
    # Make predictions by feeding model through likelihood
    with torch.no_grad(), gpytorch.settings.fast_pred_var():
        # Test points are regularly spaced along [0,1]
        return likelihood(model(test_x))
```

**Comment by arthus701** [comment]
Hm, I don't think this change is necessary. The notebook's focus is on implementing the kernel, and before `predict` is called, the models **are** set into `eval` mode...

**Comment by mboedigh** [comment]
Maybe not required strictly speaking. But it seems like it would be helpful to get as many people up and running with the basics and to make the library as accessible as possible. If there is a good way to make a general predict function, why not include it? The tutorial did say that the functions were to allow reuse (or to that effect). The examples are really good overall, but I did spend some time scratching my because of side-issues like these. Ignore it if you don't think others will appreciate the effort. It would have helped me.

**Comment by gpleiss** [comment]
```python
# Get into evaluation (predictive posterior) mode and predict
model.eval()
likelihood.eval()
observed_pred = predict(model, likelihood)
```

The model is put into `eval` mode before calling `predict`.

**Comment by arthus701** [comment]
> Maybe not required strictly speaking. But it seems like it would be helpful to get as many people up and running with the basics and to make the library as accessible as possible. If there is a good way to make a general predict function, why not include it? The tutorial did say that the functions were to allow reuse (or to that effect). The examples are really good overall, but I did spend some time scratching my because of side-issues like these. Ignore it if you don't think others will appreciate the effort. It would have helped me.

What do you think @gpleiss? It wouldn't be much work and I could certainly open a PR that implements the `predict` function the proposed way (i.e. with setting the model into `eval` mode inside of the function), but I do not want to carry the PRs for this example to excess.

**Comment by gpleiss** [comment]
@arthus701 yeah - it doesn't hurt to have the `model.eval()` statement in the `predict` function. I'm making a few improvements to the docs today, so I can implement this change.
